Docs: Fix issue about the `key_vault_secrets_provider` in "azurerm_kubernetes_cluster" is not optional like specify in documentation (#20622)
hashicorp/terraform-provider-azurerm: website/docs/r/kubernetes_cluster.html.markdown

* `secret_rotation_interval` - (Optional) The interval to poll for secret rotation. This attribute is only set when `secret_rotation` is true and defaults to `2m`.

> -> **NOTE:** When `key_vault_secrets_provider` is set - One of `secret_rotation_enabled, secret_rotation_interval` must be specified.

Reviewer: ```suggestion
-> **NOTE:** To enable`key_vault_secrets_provider` either `secret_rotation_enabled` or `secret_rotation_interval` must be specified.
```
